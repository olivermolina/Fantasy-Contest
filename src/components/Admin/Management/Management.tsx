import React from 'react';
import { styled } from '@mui/material/styles';
import MuiAccordion, { AccordionProps } from '@mui/material/Accordion';
import { ReferralCode, TransactionType, User, UserType } from '@prisma/client';
import MuiAccordionSummary, {
  AccordionSummaryProps,
} from '@mui/material/AccordionSummary';
import ArrowForwardIosSharpIcon from '@mui/icons-material/ArrowForwardIosSharp';
import MuiAccordionDetails from '@mui/material/AccordionDetails';

export interface ManagementBaseProps {
  /**
   * List of users to be selected
   */
  users: ProfileManagementUser[];
  /**
   * Boolean to show loading state in the select component
   */
  isLoading: boolean;
  /**
   * A React hook for selected userId
   */
}

export const Accordion = styled((props: AccordionProps) => (
  <MuiAccordion disableGutters elevation={0} square {...props} />
))(({ theme }) => ({
  border: `1px solid ${theme.palette.divider}`,
  '&:not(:last-child)': {
    borderBottom: 0,
  },
  '&:before': {
    display: 'none',
  },
}));
export const AccordionSummary = styled((props: AccordionSummaryProps) => (
  <MuiAccordionSummary
    expandIcon={<ArrowForwardIosSharpIcon sx={{ fontSize: '0.9rem' }} />}
    {...props}
  />
))(({ theme }) => ({
  backgroundColor:
    theme.palette.mode === 'dark'
      ? 'rgba(255, 255, 255, .05)'
      : 'rgba(0, 0, 0, .03)',
  flexDirection: 'row-reverse',
  '& .MuiAccordionSummary-expandIconWrapper.Mui-expanded': {
    transform: 'rotate(90deg)',
  },
  '& .MuiAccordionSummary-content': {
    marginLeft: theme.spacing(1),
  },
}));
export const AccordionDetails = styled(MuiAccordionDetails)(({ theme }) => ({
  padding: theme.spacing(2),
  borderTop: '1px solid rgba(0, 0, 0, .125)',
}));

export type UserManagementBaseInputs = {
  user: User | string;
};

export interface AddRemoveWithdrawableInputs extends UserManagementBaseInputs {
  amount: number;
  transactionType: TransactionType;
}

export interface FreeCreditInputs extends UserManagementBaseInputs {
  creditAmount: number;
}

export type ManagementInputs = AddRemoveWithdrawableInputs & FreeCreditInputs;

export interface ProfileManagementUser {
  id: string;
  email: string;
  username: string | null;
  referral: string | null;
  referralCodes: ReferralCode[];
  type: UserType;
}
