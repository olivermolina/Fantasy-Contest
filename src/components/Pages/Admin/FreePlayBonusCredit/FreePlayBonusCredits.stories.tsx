import React from 'react';
import { ComponentMeta, ComponentStory } from '@storybook/react';

import FreePlayBonusCreditComponent from './FreePlayBonusCredit';
import { usersMock } from '~/components/Admin/Management/__mocks__/usersMocks';

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: 'Lockspread/Admin/Management/FreePlayBonusCredit',
  component: FreePlayBonusCreditComponent,
  // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
  argTypes: {
    backgroundColor: { control: 'color' },
  },
} as ComponentMeta<typeof FreePlayBonusCreditComponent>;

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof FreePlayBonusCreditComponent> = (
  args,
) => <FreePlayBonusCreditComponent {...args} />;

export const FreePlayBonusCredit = Template.bind({});
// More on args: https://storybook.js.org/docs/react/writing-stories/args

FreePlayBonusCredit.args = {
  onSubmit: () => alert('test'),
  users: usersMock,
  isLoading: false,
};
