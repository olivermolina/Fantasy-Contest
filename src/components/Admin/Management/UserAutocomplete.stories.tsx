import React from 'react';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import UserAutocompleteComponent from './UserAutocomplete';
import { useForm } from 'react-hook-form';
import { ManagementInputs } from '~/components/Admin/Management/Management';
import { usersMock } from '~/components/Admin/Management/__mocks__/usersMocks';

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: 'Lockspread/Admin/UserManagement',
  component: UserAutocompleteComponent,
  // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
  argTypes: {
    backgroundColor: { control: 'color' },
  },
} as ComponentMeta<typeof UserAutocompleteComponent>;

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof UserAutocompleteComponent> = (args) => {
  const { control } = useForm<ManagementInputs>();
  return <UserAutocompleteComponent {...args} control={control} />;
};

export const UserAutocomplete = Template.bind({});
// More on args: https://storybook.js.org/docs/react/writing-stories/args

UserAutocomplete.args = {
  users: usersMock,
  isLoading: false,
};
