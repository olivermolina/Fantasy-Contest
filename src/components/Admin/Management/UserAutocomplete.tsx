import React from 'react';
import { Control, Controller } from 'react-hook-form';
import Autocomplete from '@mui/material/Autocomplete';
import { TextField } from '@mui/material';
import CircularProgress from '@mui/material/CircularProgress';
import { ProfileManagementUser } from '~/components/Admin/Management/Management';
import z from 'zod';

interface UserAutocompleteProps {
  /**
   * List of users to be selected
   */
  users: ProfileManagementUser[];
  /**
   * Optional react hook for selected userId
   */
  setSelectedUserId?: React.Dispatch<React.SetStateAction<string | undefined>>;
  /**
   * Boolean to show loading state in the select component
   */
  isLoading: boolean;
  /**
   * A React-Hook form control object
   */
  control: Control<any, any>;
  /**
   * Custom select label
   */
  label?: string;
  /**
   * Boolean to disable the select component
   */
  disabled?: boolean;
  defaultValue?: {
    id: string;
    username: string;
  };
}

export const UserSelectSchema = z.object({
  user: z.object(
    {
      id: z.string().optional(),
      username: z.string().optional(),
    },
    {
      required_error: 'Please select a user',
      invalid_type_error: 'Please select a user',
    },
  ),
});

const UserAutocomplete = (props: UserAutocompleteProps) => {
  const {
    users,
    setSelectedUserId,
    isLoading,
    control,
    disabled,
    defaultValue,
  } = props;
  return (
    <Controller
      render={({ field, fieldState }) => (
        <Autocomplete
          {...field}
          disablePortal
          disableClearable
          onChange={(event, newValue) => {
            if (typeof newValue === 'string') {
              // Do nothing
            } else {
              field.onChange(newValue);
              if (newValue?.id && setSelectedUserId) {
                setSelectedUserId(newValue.id);
              }
            }
          }}
          getOptionLabel={(option) => {
            // e.g value selected with enter, right from the input
            if (typeof option === 'string') {
              return option;
            }
            return option.username || option.email || '';
          }}
          id={'userId'}
          options={users || []}
          selectOnFocus
          clearOnBlur
          handleHomeEndKeys
          renderOption={(props, user) => (
            <li {...props}>{user.username || user.email}</li>
          )}
          freeSolo
          renderInput={(params) => (
            <TextField
              {...params}
              label={props.label || 'Select User'}
              fullWidth
              error={!!fieldState.error?.message}
              helperText={fieldState.error?.message}
              size={'small'}
              InputProps={{
                ...params.InputProps,
                endAdornment: (
                  <React.Fragment>
                    {isLoading ? (
                      <CircularProgress color="inherit" size={20} />
                    ) : null}
                    {params.InputProps.endAdornment}
                  </React.Fragment>
                ),
              }}
            />
          )}
          fullWidth
          disabled={disabled}
        />
      )}
      name={'user'}
      control={control}
      defaultValue={defaultValue || ''}
    />
  );
};

export default UserAutocomplete;
