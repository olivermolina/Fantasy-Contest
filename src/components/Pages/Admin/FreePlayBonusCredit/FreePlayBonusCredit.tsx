import React, { useEffect } from 'react';
import { Button, InputAdornment, TextField } from '@mui/material';
import { useForm } from 'react-hook-form';
import { useTheme } from '@mui/material/styles';
import useMediaQuery from '@mui/material/useMediaQuery';
import { yupResolver } from '@hookform/resolvers/yup';
import * as Yup from 'yup';
import UserAutocomplete from '~/components/Admin/Management/UserAutocomplete';
import {
  ManagementBaseProps,
  ManagementInputs,
} from '~/components/Admin/Management/Management';

const InputValidationSchema = Yup.object().shape({
  creditAmount: Yup.number().typeError('Please provide an amount'),
  user: Yup.object()
    .typeError('Please select user')
    .required('Please select user'),
});

interface FreePlayBonusCreditProps extends ManagementBaseProps {
  /**
   * Submit form function
   */
  onSubmit: (inputs: ManagementInputs) => void;
  setSelectedUserId: React.Dispatch<React.SetStateAction<string | undefined>>;
}

const FreePlayBonusCredit = (props: FreePlayBonusCreditProps) => {
  const { onSubmit, users, isLoading, setSelectedUserId } = props;

  const theme = useTheme();
  const matches = useMediaQuery(theme.breakpoints.down('md'));

  const {
    register,
    handleSubmit,
    formState: { errors, isSubmitSuccessful },
    reset,
    control,
  } = useForm<ManagementInputs>({
    resolver: yupResolver(InputValidationSchema),
  });

  useEffect(() => {
    if (isSubmitSuccessful) reset();
  }, [isSubmitSuccessful]);

  return (
    <div className={'flex flex-col gap-2 py-4'}>
      <form onSubmit={handleSubmit(onSubmit)}>
        <div className={`flex flex-col md:flex-row items-start gap-4`}>
          <UserAutocomplete
            control={control}
            users={users}
            isLoading={isLoading}
            setSelectedUserId={setSelectedUserId}
          />
          <TextField
            id="outlined-basic"
            label="Amount"
            variant="outlined"
            type={'number'}
            fullWidth
            InputProps={{
              startAdornment: (
                <InputAdornment position="start">$</InputAdornment>
              ),
            }}
            {...register('creditAmount')}
            error={!!errors?.creditAmount}
            helperText={errors?.creditAmount?.message}
            size={'small'}
          />
          <Button
            variant="contained"
            type="submit"
            fullWidth={matches}
            name={'submit'}
          >
            Submit
          </Button>
        </div>
      </form>
    </div>
  );
};

export default FreePlayBonusCredit;
