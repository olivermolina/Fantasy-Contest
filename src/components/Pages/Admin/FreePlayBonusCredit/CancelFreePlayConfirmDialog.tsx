import * as React from 'react';
import Button from '@mui/material/Button';
import Dialog from '@mui/material/Dialog';
import DialogActions from '@mui/material/DialogActions';
import DialogContent from '@mui/material/DialogContent';
import DialogContentText from '@mui/material/DialogContentText';
import DialogTitle from '@mui/material/DialogTitle';
import { FreePlayRowModel } from '~/components/Pages/Admin/FreePlayBonusCredit/FreePlayUserBonusCreditsTable';

interface Props {
  handleDelete: (row: FreePlayRowModel) => void;
  row?: FreePlayRowModel;
  open: boolean;
  setOpen: (open: boolean) => void;
}

export default function CancelFreePlayConfirmDialog(props: Props) {
  const { handleDelete, row, open, setOpen } = props;

  const handleClose = () => {
    setOpen(false);
  };

  const handleConfirm = () => {
    handleClose();
    if (row) {
      handleDelete(row);
    }
  };

  return (
    <Dialog
      open={open}
      onClose={handleClose}
      aria-labelledby="alert-dialog-title"
      aria-describedby="alert-dialog-description"
    >
      <DialogTitle id="alert-dialog-title">
        {'Cancel Free Play Bonus Credit?'}
      </DialogTitle>
      <DialogContent>
        <DialogContentText id="alert-dialog-description">
          Are you sure you want to free play bonus credit{' '}
          <span className={'font-semibold'}>${row?.amountBonus}</span>?
        </DialogContentText>
      </DialogContent>
      <DialogActions>
        <Button onClick={handleClose} variant={'outlined'}>
          Cancel
        </Button>
        <Button onClick={handleConfirm} variant={'contained'} autoFocus>
          Confirm
        </Button>
      </DialogActions>
    </Dialog>
  );
}
