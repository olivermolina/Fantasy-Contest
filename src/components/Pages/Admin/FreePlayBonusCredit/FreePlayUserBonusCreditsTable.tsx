import React, { useMemo, useState } from 'react';
import {
  DataGrid,
  GridColDef,
  GridRowsProp,
  GridSlotsComponentsProps,
} from '@mui/x-data-grid';
import { Button, Stack } from '@mui/material';
import CancelFreePlayConfirmDialog from '~/components/Pages/Admin/FreePlayBonusCredit/CancelFreePlayConfirmDialog';
import { TransactionType } from '@prisma/client';
import { PaymentStatusCode } from '~/constants/PaymentStatusCode';

declare module '@mui/x-data-grid' {
  interface FooterPropsOverrides {
    totalBonusCredits: number;
  }
}

type Props = {
  onSubmitDeletePlayFreeCredit(currentRow: FreePlayRowModel): unknown;
  data: readonly FreePlayRowModel[];
};

export interface FreePlayRowModel {
  id: string;
  transactionDate: Date;
  amountBonus: number;
  status: PaymentStatusCode;
  type: TransactionType;
}

function CustomFooter(props: NonNullable<GridSlotsComponentsProps['footer']>) {
  return (
    <div className={'flex p-2 font-semibold text-lg'}>
      Total Bonus Credits: {Number(props.totalBonusCredits).toFixed(2)}
    </div>
  );
}

export default function FreePlayUserBonusCreditsTable(props: Props) {
  const rows: GridRowsProp<FreePlayRowModel> = props.data;
  const [selectedRow, setSelectedRow] = useState<
    FreePlayRowModel | undefined
  >();
  const [open, setOpen] = useState(false);

  const columns: GridColDef[] = [
    { flex: 1, field: 'id', headerName: 'Transaction ID' },
    {
      flex: 1,
      field: 'transactionDate',
      headerName: 'Transaction Date',
      valueFormatter: (params) => params.value.toLocaleString(),
    },
    { flex: 1, field: 'type', headerName: 'Type' },
    { flex: 1, field: 'amountBonus', headerName: 'Amount' },
    {
      flex: 1,
      field: 'status',
      headerName: 'Status',
      renderCell: (params) => {
        const currentRow = params.row as typeof rows[0];
        return Object.keys(PaymentStatusCode)[
          Object.values(PaymentStatusCode).indexOf(currentRow.status as number)
        ];
      },
    },
    {
      flex: 1,
      field: 'action',
      headerName: 'Action',
      renderCell: (params) => {
        const currentRow = params.row as typeof rows[0];
        const onClick = () => {
          setSelectedRow(currentRow);
          setOpen(true);
        };

        return (
          <Stack direction="row" spacing={2}>
            <Button
              color="warning"
              size="small"
              onClick={onClick}
              variant={'contained'}
              disabled={currentRow.status !== PaymentStatusCode.COMPLETE}
            >
              Cancel
            </Button>
          </Stack>
        );
      },
    },
  ];

  const handleClose = () => {
    setOpen(false);
  };

  const deleteBonusCredit = (row: FreePlayRowModel) => {
    props.onSubmitDeletePlayFreeCredit(row);
    handleClose();
  };

  const totalBonusCredits = useMemo(
    () =>
      props.data.reduce(
        (acc, curr) =>
          curr.status === PaymentStatusCode.COMPLETE
            ? curr.type === TransactionType.CREDIT
              ? acc + curr.amountBonus
              : acc - curr.amountBonus
            : acc,
        0,
      ),
    [props.data],
  );

  return (
    <div className="flex flex-col gap-2">
      <span className={'text-lg font-bold'}>Bonus Credit Transactions</span>
      <div style={{ height: 300, width: '100%' }}>
        <DataGrid
          getRowId={(row) => row.id}
          rows={rows}
          columns={columns}
          slots={{
            footer: CustomFooter,
          }}
          slotProps={{
            footer: { totalBonusCredits },
          }}
        />
        <CancelFreePlayConfirmDialog
          open={open}
          setOpen={setOpen}
          handleDelete={deleteBonusCredit}
          row={selectedRow}
        />
      </div>
    </div>
  );
}
