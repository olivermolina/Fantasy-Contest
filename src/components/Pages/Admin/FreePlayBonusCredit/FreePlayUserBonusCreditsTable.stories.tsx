import React from 'react';
import { ComponentMeta, ComponentStory } from '@storybook/react';

import FreePlayUserBonusCreditsTable from './FreePlayUserBonusCreditsTable';
import { freePlayBonusCreditTableMockProps } from './__mocks__/freePlayBonusCreditTableMockProps';

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: 'Lockspread/Admin/Management/FreePlayUserBonusCreditsTable',
  component: FreePlayUserBonusCreditsTable,
  // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
  argTypes: {
    backgroundColor: { control: 'color' },
  },
} as ComponentMeta<typeof FreePlayUserBonusCreditsTable>;

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof FreePlayUserBonusCreditsTable> = (
  args,
) => <FreePlayUserBonusCreditsTable {...args} />;

export const FreePlayBonusCreditTable = Template.bind({});
// More on args: https://storybook.js.org/docs/react/writing-stories/args

FreePlayBonusCreditTable.args = freePlayBonusCreditTableMockProps;
