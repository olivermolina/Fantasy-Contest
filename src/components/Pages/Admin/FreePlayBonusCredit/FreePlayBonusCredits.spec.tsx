import { render, screen } from '@testing-library/react';
import FreePlayBonusCredit from './FreePlayBonusCredit';
import { ManagementInputs } from '~/components/Admin/Management/Management';

describe('FreeCredits', () => {
  const onSubmitMock = jest.fn((data: ManagementInputs) => {
    return Promise.resolve(data);
  });
  it('should render correctly', () => {
    const props = {
      onSubmit: onSubmitMock,
      users: [],
      isLoading: false,
      handleChange: () => () => alert('test'),
      expanded: 'credit',
      setSelectedUserId: jest.fn(),
    };
    const { container } = render(<FreePlayBonusCredit {...props} />);
    expect(container).toMatchSnapshot();
    expect(screen.getAllByText('Select User')).toBeDefined();
  });
});
