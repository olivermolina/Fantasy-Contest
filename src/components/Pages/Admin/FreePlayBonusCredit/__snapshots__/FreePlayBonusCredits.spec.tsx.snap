// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FreeCredits should render correctly 1`] = `
<div>
  <div
    class="flex flex-col gap-2 py-4"
  >
    <form>
      <div
        class="flex flex-col md:flex-row items-start gap-4"
      >
        <div
          class="MuiAutocomplete-root MuiAutocomplete-fullWidth css-1kkal6p-MuiAutocomplete-root"
          name="user"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="userId"
              id="userId-label"
            >
              Select User
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-autocomplete="list"
                aria-expanded="false"
                aria-invalid="false"
                autocapitalize="none"
                autocomplete="off"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-19qh8xo-MuiInputBase-input-MuiOutlinedInput-input"
                id="userId"
                role="combobox"
                spellcheck="false"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-yjsfm1"
                >
                  <span>
                    Select User
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for="outlined-basic"
            id="outlined-basic-label"
          >
            Amount
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-adornedStart css-8j6b76-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeSmall css-ittuaa-MuiInputAdornment-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-1pnmrwp-MuiTypography-root"
              >
                $
              </p>
            </div>
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedStart css-1ua80n0-MuiInputBase-input-MuiOutlinedInput-input"
              id="outlined-basic"
              name="creditAmount"
              type="number"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-14lo706"
              >
                <span>
                  Amount
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
          name="submit"
          tabindex="0"
          type="submit"
        >
          Submit
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </form>
  </div>
</div>
`;
